src/nvim/option.c

```diff
@@ -5947,12 +5947,13 @@ option_value2string (
   if (opp->flags & P_NUM) {
     long wc = 0;
 
-    if (wc_use_keyname(varp, &wc))
-      STRCPY(NameBuff, get_special_key_name((int)wc, 0));
-    else if (wc != 0)
-      STRCPY(NameBuff, transchar((int)wc));
-    else
+    if (wc_use_keyname(varp, &wc)) {
+      STRLCPY(NameBuff, get_special_key_name((int)wc, 0), sizeof(NameBuff));
+    } else if (wc != 0) {
+      STRLCPY(NameBuff, transchar((int)wc), sizeof(NameBuff));
+    } else {
       sprintf((char *)NameBuff, "%" PRId64, (int64_t)*(long *)varp);
```

Comment:
Linter changes adjacent lines, so you have to change this to use `snprintf` and below `/* P_STRING */` to `// P_STRING` with two spaces before `//`.